src/validators.ts

```diff
@@ -127,48 +110,35 @@ const maxLength = <E, S extends string>(
 const regex = <S extends string, E>(
   regex: RegExp,
   errorMessage: E
-): Validator<S, S, E> =>
-  validator(either.fromPredicate((s) => regex.test(s), constant(errorMessage)));
+): Validator<S, S, E> => fromPredicate((s) => regex.test(s), errorMessage);
 
 const notRegex = <S extends string, E>(
   regex: RegExp,
   errorMessage: E
-): Validator<S, S, E> =>
-  validator(
-    either.fromPredicate((s) => !regex.test(s), constant(errorMessage))
-  );
+): Validator<S, S, E> => fromPredicate((s) => !regex.test(s), errorMessage);
 
 const checked = <E>(errorMessage: E): Validator<boolean, true, E> =>
-  validator(either.fromPredicate((s): s is true => s, constant(errorMessage)));
-
-const defined = <A, E>(errorMessage: E): Validator<Option<A>, A, E> =>
-  validator(either.fromOption(constant(errorMessage)));
-
-const definedNoExtract = <A, E>(
```

Comment:
does not make sense anymore in the context of "undefined values are not wrapped in an Option"